openmrs/src/main/java/org/motechproject/openmrs/domain/Observation.java

```diff
@@ -186,10 +202,26 @@ public boolean equals(Object o) {
      * Implementation of the {@link JsonSerializer} interface for the
      * {@link Observation.ObservationValue} class.
      */
-    public static class ObservationValueSerializer implements JsonSerializer<ObservationValue> {
+    public static class ObservationValueSerializer implements JsonSerializer<ObservationValue>, JsonDeserializer<ObservationValue> {
+
         @Override
         public JsonElement serialize(ObservationValue src, Type typeOfSrc, JsonSerializationContext context) {
             return new JsonPrimitive(src.getDisplay());
         }
+
+        @Override
+        public ObservationValue deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+
+            if (json.isJsonObject()) {
```

Comment:
do we need this ? Additionally it is more readable to use only one return statement